swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        # alternative solution: 
+        # list_of_items = []
+        # for item in self.inventory:
+        #     if item.category == category:
+        #         list_of_items.append(item)
+        # return list_of_items
+
+        list_of_items = [item for item in self.inventory 
+        if item.category == category]
+        return list_of_items
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        item_being_removed = self.remove(my_item)
+        friend.add(item_being_removed)
+        print(self.inventory, friend.inventory)
+
+        item_being_removed = friend.remove(their_item)
+        self.add(item_being_removed)
+        print(self.inventory, friend.inventory)
+
+        return True
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory:
+            return False
+        my_first_item = self.inventory[0]
+        friend_first_item = friend.inventory[0]
+        return self.swap_items(friend, my_first_item, friend_first_item)
```

Comment:
Nice job reusing a method!

But here is another option if we want to consider time-space efficiency. `swap_meet` winds up being O(n) (mostly due to removing the items from the inventory). We could simply swap the items in place here to have O(1) runtime. 
```python
 self.inventory[0], friend.inventory[0] = friend.inventory[0], self.inventory[0]
```
More information on [swapping values with tuples](https://www.oreilly.com/library/view/python-cookbook/0596001673/ch01s02.html)